### PR TITLE
IEEE 802.15.4: Introduce MAC configs and drop IEEE802154_MTU constant

### PR DESCRIPTION
**Pull request by nandojve**
This intents to normalize constants and few terms accord to IEEE 802.15.4-2020. This is part of #49775 discussion.

Remarks:

* Buffer allocation should consider at least PSDU (PHY Service Data Unit) size. This means that at least a full counterpart MPDU (MAC Protocol Data Unit) should be allocated. The L2 and Upper layers are allowed to use MSDU (MAC Service Data Unit)  size. The MSDU (MAC Header + MAC Payload) is equivalent to MTU in terms of IP Protocol.
* L2 and above layers should request MTU by using `ieee802154_get_msdu_size()`. This is a constant for drivers that only support the standard 127 or 2047 bytes PSDU. However, a runtime function should be defined when multiple drivers with different PSDU sizes are required or if a MSK PHY should be used. Currently runtime capabilities are out of scope.
* It is not expected regressions since no functional changes were implemented.

CC: @fgrandel

### CODE REVIEW

File: subsys/net/l2/ieee802154/ieee802154.c [hunk 1]
```diff
@@ -347,7 +349,7 @@ static int ieee802154_send(struct net_if *iface, struct net_pkt *pkt)
 		}
 #else
 
-		if (buf->len > IEEE802154_MTU) {
+		if (buf->len > ieee802154_get_msdu_size()) {
```
Comment (cfriedt) [inline subsys/net/l2/ieee802154/ieee802154.c:352]: Shouldn't the size be a `struct *_data` area of some device? I guess that is a completely different problem though. Just a comment - I don't think that's something that can be fixed right now. The assumption everywhere in Zephyr is that there is 1 radio, but we need to facilitate multi-instance drivers.

Comment (nandojve) [reply subsys/net/l2/ieee802154/ieee802154.c:352]: Yes, you are right. I forget to pass the `iface` to be checked. I'll fix and review the others functions.

File: tests/net/ieee802154/6lo_fragment/src/main.c [hunk 2]
```diff
@@ -438,11 +438,11 @@ static struct net_fragment_data test_data_8 = {
 	.iphc = false
 };
 
-static uint8_t frame_buffer_data[IEEE802154_MTU];
+static uint8_t frame_buffer_data[CONFIG_NET_L2_IEEE802154_MPDU_SIZE];
```
Comment (fgrandel) [inline tests/net/ieee802154/6lo_fragment/src/main.c:441]: Currently this wastes space - FCS is not (yet) reported to upper layers. Should be changed only when introducing this feature in the same change set IMO.

**Review by ghost — CHANGES_REQUESTED** [review]
I have concerns regarding design/architectural choices in this PR:
- I vote against the proposed changes in user space config as they are conceptually wrong (explanation below) and will be hard to undo in the future.
- I also hope for more future-proof and standards-conforming internal APIs.

The key is: This PR seems to look at some immediate problem IMO (maybe introducing some subset of the MSK PHY) but tries to be something more generic. It would be easier if this change was framed as preparation for some subset of some PHY that is part of IEEE 802.15.4-2015+. All constants, functions etc. could then be named accordingly with this reduced objective in mind but inside a namespace that is large enough to accommodate (in principle) all IEEE 802.15.4 PHYs at some future point in time. Otherwise this change only introduces complexity without a clear objective what this will be good for in the foreseeable future.

**Comment by fgrandel** [comment]
Just for documentation: I talked this over with @nandojve in a video chat and it turned out there are no specific plans to actually contribute a standards-compliant PHY (MSK, SUN, LECIM, ...) that requires larger payloads in Zephyr in the foreseeable future. The requirement for larger packets with 4-byte checksums seems to stem from an immediate need to introduce larger packets in a non-standard/proprietary way into existing drivers - probably better done in a private driver fork or driver-specific configuration option. A change as outlined in much detail in this PR will be necessary, though, as soon as someone wants to actually introduce a standards-compliant PHY with larger packet sizes into Zephyr.

**Pull request closed without merging**